### PR TITLE
Sync Integration Test

### PR DESCRIPTION
**Pull request by AviadPepperi**
Draft as I will need:
- [x] Ask @EvgenyVendrovPepperi  to install integration test addon on all test dists.
    - [x] Asked.
    - [x] Answered.
- [x] Use new testing framework version when available ([link](https://github.com/Pepperi-Addons/addon-testing-framework/pull/18)).
- [x] [merge](https://github.com/Pepperi-Addons/User-Defined-Collections/pull/210).
- [x] Test pass without Nebula and with Sync 3.0.

### CODE REVIEW

File: server-side/integration-tests/tests/base-udc.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            token: container.client.OAuthAccessToken,
+            baseURL: container.client.BaseURL,
+            addonUUID: container.client.AddonUUID,
+            addonSecretKey: container.client.AddonSecretKey,
+            actionUUID: container.client.ActionUUID,
+        });
+    }
+
+    public getShortUUID(): string {
+        return uuidv4().split('-')[0];
```
Comment (Chasky-H) [inline server-side/integration-tests/tests/base-udc.test.ts:36]: UUIDService? or delete that service.

File: server-side/integration-tests/tests/sync.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    tests(
+        describe: (suiteTitle: string, func: () => void) => void,
+        it: (name: string, fn: Mocha.Func) => void,
+        expect: Chai.ExpectStatic,
+    ): void {
+        describe(this.title, () => {
+
+            it('Rebuild table', async () => {
+                // Rebuild the table.
+                const nebulusService = this.container.get(NebulusService);
```
Comment (Chasky-H) [inline server-side/integration-tests/tests/sync.test.ts:33]: I don't think this test should be testing internal functionality. 
The nebulus already tests this. 
The test should be testing external UDC functionality. 

Create Scheme.
Change Sync flag.
Check that Dirty is set.
Publish via UDC endpoint
Test that data is avialable in CPI (no need to test the cache).

File: server-side/integration-tests/services/nebulus.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { AddonAPIAsyncResult, CacheDeltaInput, CacheDeltaResult, NebulusRebuildInput, PapiClient } from "@pepperi-addons/papi-sdk";
+import { BaseService, ServicesContainer } from "@pepperi-addons/addon-testing-framework";
+
+export class NebulusService extends BaseService {
+
+    protected papiClient: PapiClient;
+
+    constructor(container: ServicesContainer) {
+        super(container);
+        this.papiClient = new PapiClient({
```
Comment (roiaharon-pepperi) [inline server-side/integration-tests/services/nebulus.service.ts:10]: I believe that all the logic in this service is asynchronous, then you shouldn't pass the action UUID to the papi client, it should create a new one.

**Review by roiaharon-pepperi — COMMENTED** [review]
@AviadPepperi please note that I've changed the base branch.
either make releases/1.0 as the default, or make sure to open the PRs on the correct base

File: server-side/integration-tests/tests/base-udc.test.ts (hunk 1 above)
Comment (AviadPepperi) [reply server-side/integration-tests/tests/base-udc.test.ts:36]: Deleted.

File: server-side/integration-tests/tests/sync.test.ts (hunk 2 above)
Comment (AviadPepperi) [reply server-side/integration-tests/tests/sync.test.ts]: Removed any reference to Nebulus and cache.

**Pull request merged**